### PR TITLE
Implement DictWrite hint

### LINKED ISSUE #290: DictWrite
Writes a new value to a dictionary at a specified key, and keeps track of the previous value.

Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.cairo#L39

### PR DESCRIPTION
**Pull request by har777**
Fixes #290

### CODE REVIEW

**Comment by har777** [comment]
Moving to draft till integration tests added.

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 1]
```diff
@@ -126,6 +126,86 @@ func createDictReadHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newDictReadHint(dictPtr, key, value), nil
 }
 
+// DictWrite hint updates the value of given key for given dictionary
+// while also writing the prev_value of the key to memory
+//
+// `newDictWriteHint` takes 3 operander as argument
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:132]: operanders as arguments

Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:132]: operanders as arguments

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 2]
```diff
@@ -126,6 +126,86 @@ func createDictReadHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
...
+			if err != nil {
+				return err
+			}
+			dictionaryManager, ok := ctx.ScopeManager.GetZeroDictionaryManager()
+			if !ok {
+				return fmt.Errorf("__dict_manager not in scope")
+			}
+
+			//> dict_tracker.current_ptr += ids.DictAccess.SIZE
+			err = dictionaryManager.IncrementFreeOffset(*dictPtr, 3)
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:156]: same here, dont we need to write on the memory segment dedicated to this dict the entry created when writing in the dict?

Comment (har777) [reply pkg/hintrunner/zero/zerohint_dictionaries.go:156]: See: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.cairo#L38

The `dict_write` cairo function does that work!

The `prev_value` is written to memory inside this hint though. See line `177`.

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_dictionaries.go:156]: Perfect !

**Review by TAdev0 — APPROVED** [review]

**Comment by MaksymMalicki** [comment]
Hey @har777 could you resolve the conflicts in this branch?

**Review by MaksymMalicki — APPROVED** [review]

**Pull request merged**